src/game.cpp

```diff
@@ -6107,6 +6112,11 @@ bool pet_menu( monster *z )
         return false;
     }
 
+    if( play_with_pet == choice && query_yn( _( "Spend a few minutes to play with your %s?"), pet_name.c_str() ) ) {
+        g->u.assign_activity( activity_id( "ACT_PLAY_WITH_PET" ), rng( 50, 125 ) * 100 );
+
```

Comment:
```suggestion
        g->u.activity.str_values.push_back( pet_name );
```

Reply:
Nice, this fixed it. Even tested a few times (with/without interruptions), renaming pets, etc.